FXML-638: Lower XTen Linear to LinAlg Linear

cmcgirr-amd opened the pull request:
Conversion pass for lowering the XTen Linear Operator to the corresponding LinAlg Operator.

Requires PR: 

- [x] https://github.com/ljfitz/llvm-project/pull/20

Commit:
feat(XTenToLinAlg.cpp): add XTen Linear to LinAlg Linear Conversion

converts directly down to linalg providing the correct conversion types.
also handles when bias is optional or none and creates a empty
bias tensor by default

Commit:
test(xten_to_linalg_linear.mlir): add tests for xten linear to linalg linear

provide tests to check that linear gets converted. Including when bias is none or optional

BraunPhilipp commented on lib/Conversion/XTenToLinalg.cpp:
```diff
@@ -82,8 +82,10 @@ static Value zeroInit(ArrayRef<int64_t> sizes, mlir::Type elementType, Location
+  if (atenBias.getType().isa<Torch::NoneType>() ||
+      atenBias.getType().isa<Torch::OptionalType>()) {
+    auto resultTy =
+        op->getResult(0).getType().dyn_cast<torch::Torch::BaseTensorType>();
```
`cast` and not `dyn_cast` please or check for `nullptr`.

BraunPhilipp commented:
Otherwise, LGTM. Thank you for zeroing out the bias!

Commit:
feat(ATenVisualGraph.cpp): add support for xten linear in visual graph export

update the supported operators to include linear using the same format as the aten linear operator

Commit:
Merge remote-tracking branch 'origin/christopher.lower_liner_to_xten_linear.FXML-638' into christopher.convert_xten_to_linalg_linear.FXML-638

Commit:
fix(XTenToLinalg.cpp): use cast instead of dyn_cast

cast uses a isa check to assert we are casting to the correct type
